### PR TITLE
feat(normalize): split an equal-length protein delins whose interior is unchanged

### PR DESCRIPTION
**Pull request by nh13**
Base is #1566 (`chore/partition-model-cleanups`), not `main`. Full chain at the bottom.

## What the spec requires

`recommendations/protein/delins.md:21` states the rule and `:64` states the prohibition, in the strong form:

> two variants separated by one or more amino acids should be described individually and not as a "delins".

> **NOTE**: the variant is **not** described as `p.Ser44_Trp46delinsArgLeuArg`.

That second one is a prohibition on the unsplit spelling, not a preference between two legal forms — the same kind as the ~49 "not correct" / "is not allowed" clauses in `recommendations/checklist.md`. There is no codon carve-out competing with it on this axis: `general.md:35-38` is a *reading-frame* exception, and a protein description has no codons of its own to share.

Ferro preserved both spellings. Measured before this change, on the hermetic protein fixture whose residue 45 is `Leu` and residue 989 is `Leu`, so each pair of spellings denotes one protein sequence:

| row | input | ferro before | spec / ferro now |
|---|---|---|---|
| W48 | `NP_SPECW48.1:p.Ser44_Trp46delinsArgLeuArg` | *unchanged* | `NP_SPECW48.1:p.[Ser44Arg;Trp46Arg]` |
| W59 | `NP_SPECW48.1:p.Ser988_Gln990delinsAlaLeuHis` | *unchanged* | `NP_SPECW48.1:p.[Ser988Ala;Gln990His]` |

Those two were the entire contents of `spec_worked_example_rules.rs`'s `FERRO_IS_WRONG` table, executed by a test that was red on purpose.

## The move

An **equal-length** `delins` whose reference and payload agree on at least one interior residue is cut into the maximal runs of disagreement. A one-residue run becomes a substitution (`general.md:56` ranks substitution above delins); a longer one becomes the smaller `delins` it really is.

**Equal length, and only equal length.** A span and a payload of the same length have exactly one residue-wise correspondence, so "the middle residue is unchanged" is a fact about the input. An unequal-length `delins` has no such correspondence: locating a run inside it means first choosing an alignment, which is the re-derivation the partition model exists to remove. Those are left as authored.

**The reference decides, never the payload.** The unchanged middle residue is a claim about the *reference sequence*. An accession the provider cannot serve is declined outright rather than split on the assumption that `payload[i]` is what the reference holds — `has_protein_data()` is provider-wide, so the fetch is what decides (#1131). `Xaa` on either side is declined (equality there is not a statement about identity), and so is a `Ter` in the payload: `delins.md:47` publishes `p.(Pro578_Lys579delinsLeuTer)`, so a stop-introducing change stays a `delins` carrying its `Ter`, and `coalesce_protein_adjacent_changes` already converges toward that form.

**Not routed through the nucleotide canonicalizer.** `merge::canonicalize_from_sequence` applies the edit set to a reference window and re-derives the partition from the resulting sequence. The protein axis has no apply-to-reference — `merge::cis_kind_of` returns `None` for `Protein` for exactly that reason — so the move reads the reference residues directly. `cis_kind_of` and `is_splittable_single_member` are untouched.

## Where the pass runs, and one defect that found

It runs at the whole-description seam (`normalize_core_canonical`, beside `canonicalize_from_sequence`), **not** inside `normalize_protein`. Wiring it into the per-variant path was tried first and measured; it produced two defects that every bare-description test passes straight over:

```text
p.[Ser44_Trp46delinsArgLeuArg;Ala60Gly]
  -> [NP_003997.1:p.Ala60Gly;NP_003997.1:p.[Ser44Arg;Trp46Arg]]
p.[Ser44_Trp46delinsArgLeuArg];[Ala60Gly]
  -> NP_003997.1:p.[Ser44Arg;Trp46Arg];[Ala60Gly]        (does not re-parse)
```

The first is a nested bracket carrying a repeated accession, which is not a description. The second *is* legal HGVS — the DNA axis emits `g.[A;B];[C]` and reads it back — but ferro's **protein** allele grammar accepts only single-member arms, so normalization would emit a string `parse_hgvs` refuses. That is what `FERRO_ASSERT_REPARSE` exists to catch, and its exemption list is closed on purpose.

So the move applies to a whole `p.` description and **declines for a member of one**. Both declines are pinned as tests, as is the parser gap the trans one rests on (`the_protein_allele_grammar_reads_only_single_member_arms`), so the decline stops being necessary the moment the parser closes it. Closing it also needs a decision this PR does not take: where an inner allele's `( )` marker goes once its members flatten into the outer bracket — `protein/alleles.md:34` puts it per member, `:90` puts it round the group.

## Tests

- W48 and W59 move from `FERRO_IS_WRONG` into `WORKED`, each gaining a `NEGATIVE` row naming the exact unsplit `delins` it must not fall back to and a `PARTITIONS` row pinning the two members read off the *parsed* output.
- The red-on-purpose test becomes `the_protein_axis_splits_a_separation_one_delins`, a **convergence** guard: both spellings settle on the split form, and separation 0 still merges (`p.[Trp24Cys;Val25Arg]` -> `p.Trp24_Val25delinsCysArg`). The per-row tests each read one direction; this reads both, so a split move that also re-merged would satisfy them and fail this.
- `FERRO_IS_WRONG` is **removed**, not left empty — a test iterating an empty table passes vacuously, which that file's own policy forbids. The census pin drops from `(2, 1, 2)` to `(2, 1)`, which is a deliberate edit rather than a silent one.
- New `tests/it/protein_axis_split_move.rs`, 15 tests, covering the boundary the spec prints no example for. Positive: separation 1, separation 2, a two-residue run beside a substitution, the predicted `( )` wrapper, and a fixed-point check on every answer. Negative, each naming the exact forbidden string: unequal length, no protein data at all, protein data for a different accession, a `Ter` payload, a fully-changed run, an adjacent pair, and the two allele-member declines.

Nothing was re-blessed and no assertion was weakened.

## Verification — SUPERSEDED BY THE 2026-08-09 RESTACK, re-measurement pending

The table below was taken against **`1ae51110`**, which was #1566's head *before* the restack. That head no longer exists: #1566 is now `95e283e8`, rebased onto #1565's current generation, and this branch was rebased onto it. **The 46 -> 45 delta below therefore describes a parent that is gone**, and is kept only as the record of what was measured when. It needs re-running against `95e283e8` before this leaves draft. What the delta established — one row fixed, zero new — is the claim to re-establish, not to assume.

| | failures |
|---|---|
| baseline (`1ae51110`, superseded, `--release`) | 46 |
| this branch at that time (`--release`) | 45 |
| this branch at that time, `FERRO_ASSERT_{IDEMPOTENT,REPARSE,IN_BOUNDS}=1`, debug | 45 |

Set difference in both directions was exactly one row fixed — `spec_worked_example_rules::the_protein_axis_must_split_a_separation_one_delins`, the red-on-purpose row this PR closes — and **zero new**. The three sibling rows in that file (`every_worked_example_produces_the_spec_published_answer`, `the_forbidden_description_is_never_what_ferro_emits`, `the_normal_forms_assert_the_partitions_the_spec_asserts`) were red from the separate #1484 interaction, all three reporting only `[W58] LRG_199t1:c.[992_1002del;1004T>C]` — a row that #1570 has since fixed further up the stack, which is one reason the re-measurement is expected to land on different numbers rather than the same ones.

`cargo check --features dev --all-targets`, `cargo fmt --all --check`, `cargo clippy --all-features --all-targets` all clean at the time of that run.

## Blast radius

**Read the corpus numbers below as an upper bound on a shape, not as a measured move count** — and note first that the two committed harnesses are structurally blind here, which is why no `0 moved` figure is quoted:

- `examples/dump_normalized_corpus.rs` has **no protein axis** (`g`, `c`, `cx` only, no protein provider), so it cannot exhibit the property at all.
- The local prepared reference carries **`protein_fastas: []`**, so running the candidate rows through `ferro normalize --reference` reports `changed=0` for a reason that has nothing to do with this change.

What *can* be counted is how many rows are even in reach: an equal-length protein `delins` spanning ≥ 3 residues. Everything else on the protein axis, and the whole DNA/RNA side, is untouchable by this pass.

| committed corpus | protein `delins` ranges | equal-length | equal-length, span ≥ 3 |
|---|---|---|---|
| `clinvar_hgvs_unique` | 180 | 51 | **4** |
| `clinvar_hgvs_500k` | 414 | 91 | **5** |
| `cmrg_genes_exhaustive` | 5,553 | 1,519 | **190** |
| `paraphase_genes_exhaustive` | 1,139 | 359 | **36** |
| total | 7,286 | 2,020 | **235** |

So ~3.2 % of protein `delins` range descriptions are in reach, and each of those additionally needs an interior residue that matches its reference — which for a *derived* consequence (the CMRG/Paraphase rows) means the consequence was over-merged, i.e. the thing this rule fixes.

The evidence that carried weight was the suite: 9,768 tests, 45 failures, the set byte-identical to baseline minus this PR's own row, with **nothing re-blessed**. For an output-moving change that is the strong signal — every existing expectation already agreed with the new answer, and the only strings that moved are the two the spec publishes. That evidence is on the superseded parent and is being re-taken.

## Summary by CodeRabbit

* **New Features**
  * Protein-level replacement variants can now be automatically separated into distinct substitutions or smaller replacements when unchanged residues clearly divide multiple changed regions.
  * Normalized descriptions preserve allele and termination semantics where applicable.

* **Bug Fixes**
  * Improved consistency between equivalent protein replacement and substitution spellings.
  * Variants that are ambiguous, unsupported, contiguous, unequal in length, or contain stop codons remain unchanged rather than being incorrectly split.

* **Tests**
  * Expanded coverage for normalization, allele syntax, edge cases, and worked examples.

### Stacked PR Chain: partition-normalizer

Seven PRs, one arc: implement the partitioner, measure it, fix it with the measurements. Verified mechanically on 2026-08-09 with `git merge-base --is-ancestor` across all six links, after a restack that moved every head except #1565's — so any measurement in these descriptions taken before that restack is flagged where it appears rather than left reading as current.

| PR | Title |  Merges Into  |
|:--:|:------|:-------------:|
|#1565|feat(normalize): make the partition rule govern every block|**N/A** (`main`)|
|#1566|chore(normalize): restate the comment the partition rule made false|#1565|
|#1568|feat(normalize): split an equal-length protein delins whose interior is unchanged|#1566|
|#1567|feat(equivalence): expose a groupable SPDI key for bucketing by denoted bases|#1568|
|#1570|test(normalize): adjudicate the rows the partition model moved|#1567|
|#1571|fix(normalize): typing may not widen a member past its asserted span|#1570|
|#1572|feat(conformance): an exhaustive spec-derived corpus, and the burn-down it measures|#1571|

Merge order is top to bottom. #1565 owns the whole stack's representation move — it carries the `FERRO_PARTITION` default flip to `preserve`, and `FERRO_PARTITION=live` restores every pinned census exactly.

Representation-Change: protein axis only, split direction. An equal-length `p.` delins spanning >= 3 residues whose interior matches the reference now renders as a cis allele of the changes it describes (`p.Ser44_Trp46delinsArgLeuArg` -> `p.[Ser44Arg;Trp46Arg]`); the reverse merge and every other axis are unchanged. 2 rows move in the committed spec corpus, both to the form the spec publishes. Upper bound of ~235 rows across the four committed real corpora (7,286 protein delins ranges), each additionally requiring an interior reference match; not measurable locally because the prepared reference carries no protein FASTAs and the dump harness has no protein axis. Previously-accepted inputs, so a real migration for anyone storing protein delins strings. This is a per-PR declaration and is independent of the stack-wide move #1565 declares; the two do not overlap, since #1565's figures are DNA/transcript axes only.

### CODE REVIEW

**Comment by nh13** [comment]
@coderabbitai pause

**Comment by nh13** [comment]
@coderabbitai review

File: src/normalize/mod.rs [hunk 1]
```diff
@@ -6243,6 +6299,161 @@ impl<P: ReferenceProvider> Normalizer<P> {
...
+        // The reference residues, or nothing. Never the payload's own middle.
+        if !self.provider.has_protein_data() {
+            return None;
+        }
+        let accession = variant.accession.transcript_accession();
+        let ref_aas =
+            self.fetch_protein_window(&accession, start_pos.number - 1, end_pos.number)?;
+        if ref_aas.len() != span || ref_aas.contains(&AminoAcid::Xaa) {
+            return None;
+        }
```
[the remark this answers is not in the document]
Comment (nh13) [reply src/normalize/mod.rs:6415]: Fixed in `10f0bfa8`, and confirmed by reproducing it first.

The asymmetry was exactly as described: the payload was checked for both `Ter` and `Xaa`, while the reference span was checked for `Xaa` only.

```rust
if seq.0.contains(&AminoAcid::Ter) || seq.0.contains(&AminoAcid::Xaa) { return None; }   // payload
if ref_aas.len() != span || ref_aas.contains(&AminoAcid::Xaa) { return None; }           // reference — no Ter
```

`ref_aas.contains(&AminoAcid::Ter)` now joins the second guard.

**The authority is stronger than "prefer `Extension`".** `protein/extension.md:18` ranks the descriptions outright — *"prioritisation: (1) extension, (2) frameshift or deletion-insertion"* — and `:30` publishes the form for exactly this event: `p.Ter110GlnextTer17`, *"a variant in the stop codon (`Ter` / `*`) at position 110, changing it to a `Gln`-codon (a no-stop variant)"*. So a residue replacing a stop is described as an extension, and the split was emitting a substitution.

**Reproduced before fixing, with the guard reverted and nothing else changed:**

```
NP_003997.1:p.Ser44_Ter46delinsAlaLeuHis  ->  NP_003997.1:p.[Ser44Ala;Ter46His]
```

That is your `p.[Ser988Ala;Ter990His]` on a hermetic reference whose residue 46 is the stop.

New test `a_reference_stop_is_not_split_into_a_substitution`, plus a `stop_at_46_protein()` fixture. Worth noting why it needed its own: **the payload guard cannot catch this shape** — the payload here (`AlaLeuHis`) carries no `Ter` at all, so every existing `Ter` test passes while the defect is live.

**Blast radius measured rather than assumed.** Full `tests/it` run, guard on vs off, same tree otherwise:

```
without the guard: 46 failures
with the guard:    45 failures
fixed:   protein_axis_split_move::a_reference_stop_is_not_split_into_a_substitution
newly broken: (none)
```

The other 45 are the partition-model pins this stack inherits from #1565 and adjudicates in #1570; they are identical on both arms. `cargo fmt --check` and `cargo clippy --features dev --all-targets` clean.

Docs updated in both places that stated the old, payload-only rule: the `# When it declines` list on `try_protein_split_delins`, and the module header of `protein_axis_split_move.rs`.

File: src/normalize/mod.rs [hunk 2]
```diff
@@ -6243,6 +6299,161 @@ impl<P: ReferenceProvider> Normalizer<P> {
...
+    ///   3' of it, which `delins.md:45` forbids ("amino acids after the
+    ///   translation termination codon are **not** listed"). This is the mirror
+    ///   of the `first_ter_pos` gate in
+    ///   [`merge::coalesce_protein_adjacent_changes`](crate::normalize::merge);
+    /// - fewer than two changed runs survive, i.e. the change really is one
+    ///   contiguous `delins`.
+    fn try_protein_split_delins(
+        &self,
+        variant: &crate::hgvs::variant::ProteinVariant,
+    ) -> Option<Vec<crate::hgvs::variant::ProteinVariant>> {
```
[the remark this answers is not in the document]
Comment (nh13) [reply src/normalize/mod.rs:6373]: Not applying — the trailer is already there.

The PR description carries it at line 91, and it is a substantive disclosure rather than a decline:

> `Representation-Change: protein axis only, split direction. An equal-length `p.` delins spanning >= 3 residues whose interior matches the reference now renders as a cis allele of the changes it describes (`p.Ser44_Trp46delinsArgLeuArg` -> `p.[Ser44Arg;Trp46Arg]`); the reverse merge and every other axis are unchanged. 2 rows move in the committed spec corpus, both to the form the spec publishes. …`

The finding says *"The PR description **as summarized** carries no trailer"* — which is the tell: the summary was the input, not the description. Worth flagging as a false-positive shape, since the guideline it cites is real and the check that enforces it (`Representation change declared`) is passing on this PR.

Resolving, as there is nothing to change.

**Comment by nh13** [comment]
Superseded by #1606, which is this fix rebased onto `main` off the withdrawn partition stack, plus the open defect this PR carried.

`src/normalize/mod.rs` is byte-identical between `main` and this PR's base, so the fix applied verbatim and takes the W1–W95 suite from 16/18 to 17/18 on `main`.

Two additions. The **residue-1 guard**: `p.Met1_Ala3delinsValAlaTrp` has the W48 shape exactly, so every other gate passes it through, and the members it produces are `p.[Met1Val;Ala3Trp]` — which `protein/substitution.md:49` forbids and `parse_hgvs` refuses. Measured on `main`, that input is currently returned unchanged and re-parses, so this PR as it stands would *introduce* an unparseable output. The guard is scoped to residue 1 **changing**, not to any span reaching it, with a test pinning that.

And the reason the two files could not be split into separate PRs: this PR's own version of `spec_worked_example_rules.rs` deletes the red-on-purpose protein test, so the suite is only green with the fix present.

Separately, #1607 records that a *declined* split still emits an illegal `p.Met1Val` on `main` via affix trimming in `try_protein_delins_canonicalize`. That one is pre-existing and untouched by either PR.
